lib/search.rb

```diff
@@ -1106,7 +1108,11 @@ def self.to_tsquery(ts_config: nil, term:, joiner: nil)
   end
 
   def self.set_tsquery_weight_filter(term, weight_filter)
+    # The term is nested under 2 layers of single quotes so we need to escape
+    # single quotes and backslashs manually here to avoid generating SQL with
+    # invalid syntax.
     term = term.gsub("'", "''")
+    term.gsub!('\\', '\\\\\\\\')
```

Comment:
There is a `QUOTE_LITERAL` function in postgres but because we have two layers of single quotes nesting here, it doesn't really work well since we have to call it twice. Plus, it makes the query so much harder to understand.